hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/PrngSystemContract.java

```diff
@@ -74,6 +84,7 @@ public FullResult computeFully(@NonNull final Bytes input, @NonNull final Messag
         final ContractID contractID = asEvmContractId(Address.fromHexString(PRNG_PRECOMPILE_ADDRESS));
 
         try {
+            validateTrue(frame.getRemainingGas() >= gasRequirement, INSUFFICIENT_GAS);
```

Comment:
We shouldn't need this, should we?

I'd expect [this code path](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/processors/CustomMessageCallProcessor.java#L181) to already take care of that.